module/data/shared/roll-config-field.mjs

```diff
@@ -0,0 +1,24 @@
+import {FormulaField} from "../fields.mjs";
+
+const { StringField, NumberField, SchemaField } = foundry.data.fields;
+
+/**
+ * Field for storing data for a specific type of roll.
+ */
+export default class RollConfigField extends foundry.data.fields.SchemaField {
+  constructor({roll={}, ...fields}={}, options={}) {
+    const opts = { initial: null, nullable: true, min: 1, max: 20, integer: true };
+    fields = {
+      ability: new StringField({required: true, label: "DND5E.AbilityModifier"}),
```

Comment:
I think we need a way to pass in a default value for the ability. Maybe something like this:

```suggestion
  constructor({ability="", roll={}, ...fields}={}, options={}) {
    const opts = { initial: null, nullable: true, min: 1, max: 20, integer: true };
    fields = {
      ability: new StringField({required: true, initial: ability, label: "DND5E.AbilityModifier"}),
```